projects/vdk-core/src/vdk/internal/builtin_plugins/run/file_based_step.py

```diff
@@ -146,7 +146,7 @@ def invoke_run_function(func: Callable, job_input: IJobInput, step_name: str):
                     what_happened=f"Data Job step {step_name} completed with error.",
                     why_it_happened=errors.MSG_WHY_FROM_EXCEPTION(e),
                     consequences="I will not process the remaining steps (if any), "
-                    "and this Data Job execution will be marked as failed.",
```

Comment:
I didn't notice this. That's why I dislike the exception approach for control flow (as i mentioned before). 

Isn't it better to catch teh exception and handle it, re-raise it. I think this will print an error in the logs - but we have not error ?